dataset.py

```diff
@@ -108,6 +108,33 @@ def random_subseq(sequence, seq_len):
     index = random.randint(0, len(sequence) - 1 - seq_len)
     return sequence[index:index + seq_len]
 
+def stretch_sequence(sequence, stretch_multiplier):
+    """ Iterate through sequence and stretch each time shift event by a factor """
+    time_sum = 0
+    time_count = 0
+    i = 0
+    while i < len(sequence):
+        evt = sequence[i]
+        if evt >= TIME_OFFSET and evt < VEL_OFFSET:
+            # Convert time event to number of seconds
+            time_sum += convert_time_evt_to_sec(evt)
+            time_count += 1
+            # TODO: Edge case where last event is time shift event
+        else:
+            if i > 0:
+                # Once there is a non time shift event, go backwards and recalculate previous sequential time shift events
+                for j in range(i - 1, i - 1 -time_count, -1):
+                    stretch_time = (convert_time_evt_to_sec(sequence[j]) / time_sum) * (stretch_multiplier * time_sum)
+                    stretch_ticks = round(stretch_time * TICKS_PER_SEC)
+                    tick_bin = find_tick_bin(stretch_ticks)
+                    # Reassign event with 2x slower time
+                    sequence[j] = TIME_OFFSET + tick_bin
+                # Reset tracking variables
+                time_sum = 0
+                time_count = 0
+        i += 1
+    return sequence
```

Comment:
The sequence length is not guaranteed to be kept constant after the time shift operation. Some time shifts may incur additional events. So you will need to slice the end of a sequence to ensure the # of events is constant.

Reply:
I modify time shift events in place so I'm not sure how additional events would be created.